### ISSUE TITLE
Upgrade from 5.2.1 to 5.3.0 causes database issues.

### ISSUE THREAD

**Issue by troycarpenter** [ampache/ampache#3221]
## Description
Album Artists returns "No Artist Found" in web interface.   I suspected this would be an issue based on my past experience with upgrading to the latest release.

### Describe the bug
There appears to be a problem during the database upgrades.  However, I'm sure this also happens when installing from scratch on this release.

### To reproduce
I'm not sure if the upgrade from 5.2.1 makes a difference.

1. Docker pull latest version (at least version with digest f43ba4ba9a2b) and run it.
2. Web interface shows a number of database upgrades to be done (this step may have produced errors but the log 
3. When finished continue to the Ampache page.
4. In the left menu, under Browse, click "Album Artists" (catalog filter does not matter)
5. Results in the mid frame shows "No Artist found"

### Expected behavior
I expected all the album artists listed per the current catalog filter.

## Environment

* Ampache version: docker latest (shows f43ba4ba9a2b).  5.3.0
* Server operating system: Linux docker host
* Client operating system: Windows

### Client type

* Web

### Logs from database upgrade:
2022-04-20T10:46:33-05:00 [] (Ampache\Module\Application\ApplicationRunner) -> Found handler "Ampache\Module\Application\Update\UpdateAction" for action "update"
2022-04-20T10:46:33-05:00 [] (Ampache\Module\System\Update) -> run_update: starting
2022-04-20T10:46:33-05:00 [] (Ampache\Module\System\Update) -> run_update: checking versions
2022-04-20T10:46:34-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530000
2022-04-20T10:46:35-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530001
2022-04-20T10:46:36-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530002
2022-04-20T10:46:36-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530003
2022-04-20T10:46:36-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530004
2022-04-20T10:46:36-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530005
2022-04-20T10:46:36-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530006
2022-04-20T10:46:37-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530007
2022-04-20T10:46:37-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530008
2022-04-20T10:46:37-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530009
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530010
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530011
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_full_index`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_full_index`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_full_index`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_full_index`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_full_index`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_full_index`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_full_index`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_full_index`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_type_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_type_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_type_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_type_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_type_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_type_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_type_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_type_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_date_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_date_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_date_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_date_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_date_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_date_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_date_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_date_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_user_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_user_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_user_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_user_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_user_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_user_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_user_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_user_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_unique`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_unique`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_unique`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_unique`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_unique`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_unique`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_unique`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_unique`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530012
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530013
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_UNIQUE_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_UNIQUE_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_UNIQUE_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_UNIQUE_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_UNIQUE_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_UNIQUE_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: ALTER TABLE `object_count` DROP KEY `object_count_UNIQUE_IDX`; []
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1091,"Can't DROP INDEX `object_count_UNIQUE_IDX`; check that it exists"]
2022-04-20T10:46:38-05:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:46:46-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530014
2022-04-20T10:46:46-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530015
2022-04-20T10:46:46-05:00 [] (Ampache\Module\System\Update) -> run_update: successfully updated to 530016
2022-04-20T10:46:46-05:00 [] (Ampache\Module\System\Update) -> run_update: starting rebuild_all_preferences
2022-04-20T10:46:47-05:00 [] (Ampache\Module\System\Update) -> run_update: Upgrade complete

After that, started seeing database errors like this:
2022-04-20T10:46:56-05:00 [ampache] (Ampache\Module\System\Dba) -> Error_query SQL: SELECT DISTINCT `album`.`id`, MAX(`album`.`disk`) AS `disk` FROM `album` LEFT JOIN `song` ON `song`.`album`=`album`.`id`
 LEFT JOIN `catalog` ON `catalog`.`id` = `song`.`catalog` WHERE (`album`.`name` = ? OR LTRIM(CONCAT(COALESCE(`album`.`prefix`, ''), ' ', `album`.`name`)) = ? ) AND `album`.`mbid` IS NULL AND `album`.`mbid
_group` IS NULL AND `album`.`album_artist` = ? AND `album`.`catalog` = ?  AND `catalog`.`enabled` = '1' AND `album`.`id` IN (SELECT `object_id` FROM `catalog_map` LEFT JOIN `catalog` ON `catalog_map`.`cat
alog_id` = `catalog`.`id` WHERE `catalog_map`.`object_type` = 'album' AND `catalog`.`filter_user` IN (0, ) GROUP BY `catalog_map`.`object_id`)  GROUP BY `album`.`id` ORDER BY `disk` ASC ["American IV: The
 Man Comes Around","American IV: The Man Comes Around","480","7"]
2022-04-20T10:46:56-05:00 [ampache] (Ampache\Module\System\Dba) -> Error_query MSG: ["42000",1064,"You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version fo
r the right syntax to use near ') GROUP BY `catalog_map`.`object_id`)  GROUP BY `album`.`id` ORDER BY `disk` ASC' at line 1"]

### Ampache Logs from Album Artist display
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\Application\ApplicationRunner) -> Found handler "Ampache\Module\Application\Browse\AlbumArtistAction" for action "album_artist"
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Session) -> Session created: 72bc5cd1decfdb54549e67136de5312d
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query SQL: SELECT `artist`.`id` FROM `artist` LEFT JOIN `catalog_map` ON `catalog_map`.`object_id` = `artist`.`id` AND `catalog_map`.`object_type` = 'artist' LEFT JOIN `catalog` ON `catalog`.`id` = `catalog_map`.`catalog_id` WHERE `catalog`.`enabled` = '1' AND  `id` IN (SELECT `artist_id` FROM `artist_map` WHERE `artist_map`.`object_type` = 'album') AND  `artist`.`id` IN (SELECT `object_id` FROM `catalog_map` LEFT JOIN `catalog` ON `catalog_map`.`catalog_id` = `catalog`.`id` WHERE `catalog_map`.`object_type` = 'artist' AND `catalog`.`filter_user` IN (0, 2) GROUP BY `catalog_map`.`object_id`)  GROUP BY `artist`.`name`, `artist`.`id` ORDER BY `artist`.`name` ASC LIMIT 0, 50 []
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query MSG: ["23000",1052,"Column 'id' in IN\/ALL\/ANY subquery is ambiguous"]
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query SQL: SELECT `artist`.`id` FROM `artist` LEFT JOIN `catalog_map` ON `catalog_map`.`object_id` = `artist`.`id` AND `catalog_map`.`object_type` = 'artist' LEFT JOIN `catalog` ON `catalog`.`id` = `catalog_map`.`catalog_id` WHERE `catalog`.`enabled` = '1' AND  `id` IN (SELECT `artist_id` FROM `artist_map` WHERE `artist_map`.`object_type` = 'album') AND  `artist`.`id` IN (SELECT `object_id` FROM `catalog_map` LEFT JOIN `catalog` ON `catalog_map`.`catalog_id` = `catalog`.`id` WHERE `catalog_map`.`object_type` = 'artist' AND `catalog`.`filter_user` IN (0, 2) GROUP BY `catalog_map`.`object_id`)  GROUP BY `artist`.`name`, `artist`.`id` ORDER BY `artist`.`name` ASC LIMIT 0, 50 []
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query MSG: ["23000",1052,"Column 'id' in IN\/ALL\/ANY subquery is ambiguous"]
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query SQL: SELECT `artist`.`id` FROM `artist` LEFT JOIN `catalog_map` ON `catalog_map`.`object_id` = `artist`.`id` AND `catalog_map`.`object_type` = 'artist' LEFT JOIN `catalog` ON `catalog`.`id` = `catalog_map`.`catalog_id` WHERE `catalog`.`enabled` = '1' AND  `id` IN (SELECT `artist_id` FROM `artist_map` WHERE `artist_map`.`object_type` = 'album') AND  `artist`.`id` IN (SELECT `object_id` FROM `catalog_map` LEFT JOIN `catalog` ON `catalog_map`.`catalog_id` = `catalog`.`id` WHERE `catalog_map`.`object_type` = 'artist' AND `catalog`.`filter_user` IN (0, 2) GROUP BY `catalog_map`.`object_id`)  GROUP BY `artist`.`name`, `artist`.`id` ORDER BY `artist`.`name` ASC LIMIT 0, 50 []
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query MSG: ["23000",1052,"Column 'id' in IN\/ALL\/ANY subquery is ambiguous"]
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query SQL: SELECT `artist`.`id` FROM `artist` LEFT JOIN `catalog_map` ON `catalog_map`.`object_id` = `artist`.`id` AND `catalog_map`.`object_type` = 'artist' LEFT JOIN `catalog` ON `catalog`.`id` = `catalog_map`.`catalog_id` WHERE `catalog`.`enabled` = '1' AND  `id` IN (SELECT `artist_id` FROM `artist_map` WHERE `artist_map`.`object_type` = 'album') AND  `artist`.`id` IN (SELECT `object_id` FROM `catalog_map` LEFT JOIN `catalog` ON `catalog_map`.`catalog_id` = `catalog`.`id` WHERE `catalog_map`.`object_type` = 'artist' AND `catalog`.`filter_user` IN (0, 2) GROUP BY `catalog_map`.`object_id`)  GROUP BY `artist`.`name`, `artist`.`id` ORDER BY `artist`.`name` ASC LIMIT 0, 50 []
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query MSG: ["23000",1052,"Column 'id' in IN\/ALL\/ANY subquery is ambiguous"]
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query SQL: SELECT `artist`.`id` FROM `artist` LEFT JOIN `catalog_map` ON `catalog_map`.`object_id` = `artist`.`id` AND `catalog_map`.`object_type` = 'artist' LEFT JOIN `catalog` ON `catalog`.`id` = `catalog_map`.`catalog_id` WHERE `catalog`.`enabled` = '1' AND  `id` IN (SELECT `artist_id` FROM `artist_map` WHERE `artist_map`.`object_type` = 'album') AND  `artist`.`id` IN (SELECT `object_id` FROM `catalog_map` LEFT JOIN `catalog` ON `catalog_map`.`catalog_id` = `catalog`.`id` WHERE `catalog_map`.`object_type` = 'artist' AND `catalog`.`filter_user` IN (0, 2) GROUP BY `catalog_map`.`object_id`)  GROUP BY `artist`.`name`, `artist`.`id` ORDER BY `artist`.`name` ASC LIMIT 0, 50 []
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query MSG: ["23000",1052,"Column 'id' in IN\/ALL\/ANY subquery is ambiguous"]
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query SQL: SELECT `artist`.`id` FROM `artist` LEFT JOIN `catalog_map` ON `catalog_map`.`object_id` = `artist`.`id` AND `catalog_map`.`object_type` = 'artist' LEFT JOIN `catalog` ON `catalog`.`id` = `catalog_map`.`catalog_id` WHERE `catalog`.`enabled` = '1' AND  `id` IN (SELECT `artist_id` FROM `artist_map` WHERE `artist_map`.`object_type` = 'album') AND  `artist`.`id` IN (SELECT `object_id` FROM `catalog_map` LEFT JOIN `catalog` ON `catalog_map`.`catalog_id` = `catalog`.`id` WHERE `catalog_map`.`object_type` = 'artist' AND `catalog`.`filter_user` IN (0, 2) GROUP BY `catalog_map`.`object_id`)  GROUP BY `artist`.`name`, `artist`.`id` ORDER BY `artist`.`name` ASC LIMIT 0, 50 []
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query MSG: ["23000",1052,"Column 'id' in IN\/ALL\/ANY subquery is ambiguous"]
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query SQL: SELECT `artist`.`id` FROM `artist` LEFT JOIN `catalog_map` ON `catalog_map`.`object_id` = `artist`.`id` AND `catalog_map`.`object_type` = 'artist' LEFT JOIN `catalog` ON `catalog`.`id` = `catalog_map`.`catalog_id` WHERE `catalog`.`enabled` = '1' AND  `id` IN (SELECT `artist_id` FROM `artist_map` WHERE `artist_map`.`object_type` = 'album') AND  `artist`.`id` IN (SELECT `object_id` FROM `catalog_map` LEFT JOIN `catalog` ON `catalog_map`.`catalog_id` = `catalog`.`id` WHERE `catalog_map`.`object_type` = 'artist' AND `catalog`.`filter_user` IN (0, 2) GROUP BY `catalog_map`.`object_id`)  GROUP BY `artist`.`name`, `artist`.`id` ORDER BY `artist`.`name` ASC LIMIT 0, 50 []
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query MSG: ["23000",1052,"Column 'id' in IN\/ALL\/ANY subquery is ambiguous"]
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Database connection...
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query SQL: SELECT `artist`.`id` FROM `artist` LEFT JOIN `catalog_map` ON `catalog_map`.`object_id` = `artist`.`id` AND `catalog_map`.`object_type` = 'artist' LEFT JOIN `catalog` ON `catalog`.`id` = `catalog_map`.`catalog_id` WHERE `catalog`.`enabled` = '1' AND  `id` IN (SELECT `artist_id` FROM `artist_map` WHERE `artist_map`.`object_type` = 'album') AND  `artist`.`id` IN (SELECT `object_id` FROM `catalog_map` LEFT JOIN `catalog` ON `catalog_map`.`catalog_id` = `catalog`.`id` WHERE `catalog_map`.`object_type` = 'artist' AND `catalog`.`filter_user` IN (0, 2) GROUP BY `catalog_map`.`object_id`)  GROUP BY `artist`.`name`, `artist`.`id` ORDER BY `artist`.`name` ASC LIMIT 0, 50 []
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Module\System\Dba) -> Error_query MSG: ["23000",1052,"Column 'id' in IN\/ALL\/ANY subquery is ambiguous"]
2022-04-20T10:54:19-05:00 [user@ampache] (Ampache\Repository\Model\Browse) -> Show objects called for type {artist}

**Comment by kuzi-moto** [comment]
Are you using the standard image, or the `nosql` tag? I use `nosql`, with a mariadb 10.5 container and did not run into this problem.

**Comment by troycarpenter** [comment]
I am running the standard image with the `latest` tag.  The database is part of the image.

**Comment by troycarpenter** [comment]
So took a fresh docker install of the image tagged `develop`.  I added a single catalog.  When it was done, I hit "albums" in the browse section.  It looks like all the albums displayed, but I got this SQL error for every album in the catalog:

2022-04-20T15:52:33-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ') GROUP BY `catalog_map`.`object_id`)  GROUP BY `album`.`id` ORDER BY `disk` ASC' at line 1
2022-04-20T15:52:33-05:00 [] (Ampache\Module\System\Dba) -> Error_query SQL: SELECT DISTINCT `album`.`id`, MAX(`album`.`disk`) AS `disk` FROM `album` LEFT JOIN `song` ON `song`.`album`=`album`.`id` LEFT JOIN `catalog` ON `catalog`.`id` = `song`.`catalog` WHERE (`album`.`name` = ? OR LTRIM(CONCAT(COALESCE(`album`.`prefix`, ''), ' ', `album`.`name`)) = ? ) AND `album`.`mbid` IS NULL AND `album`.`mbid_group` IS NULL AND `album`.`album_artist` = ? AND `album`.`catalog` = ?  AND `catalog`.`enabled` = '1' AND `album`.`id` IN (SELECT `object_id` FROM `catalog_map` LEFT JOIN `catalog` ON `catalog_map`.`catalog_id` = `catalog`.`id` WHERE `catalog_map`.`object_type` = 'album' AND `catalog`.`filter_user` IN (0, ) GROUP BY `catalog_map`.`object_id`)  GROUP BY `album`.`id` ORDER BY `disk` ASC ["Visions","Visions","3","1"]
2022-04-20T15:52:33-05:00 [] (Ampache\Module\System\Dba) -> Error_query MSG: SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ') GROUP BY `catalog_map`.`object_id`)  GROUP BY `album`.`id` ORDER BY `disk` ASC' at line 1


I did not get the Album Artist problem (I noticed that in Ampache develop, "Album Artists" simply became "Artists").  I am adding another catalog to see if that's when the problem arises.

**Comment by lachlan-00** [comment]
that's the error here
```
AND id IN 
```

**Comment by lachlan-00** [comment]
3f056722d sets that

**Comment by lachlan-00** [comment]
i'll rebuild develop/latest with any extra fixes today

**Comment by troycarpenter** [comment]
Thanks for that.  I will try the new one when it is available.  I'm trying to get this to a point where I can roll it out to my family and replace Airsonic-Advanced.

On the original issue, using the docker develop image from scratch.  Continuing the test from above, I added a second catalog to the database and this time I didn't get the SQL errors with two catalogs in the GUI.  I will add some more and see what happens.

However, the Subsonic API error as mentioned #3222 still happens.

**Comment by troycarpenter** [comment]
I added a third catalog which has artists which overlap with the other catalogs.  The original SQL error still has not shown up.  I am now going to load the latest development docker image to see if the album SQL error has been fixed.

**Comment by troycarpenter** [comment]
Using the latest docker develop I am down to one SQL error when an album is displayed.  I will open a new bug for that error.  The original SQL error with the artists has been fixed.

I went back to my original install (the one with all the database upgrades) and ran the current docker development image against my original database and the artist SQL errors do not happen on that install anymore.

I have opened up #3224 to cover the second set of SQL errors that appear when displaying an album.
